README.md

```diff
@@ -7,7 +7,112 @@
 [![crates.io](https://img.shields.io/crates/v/stm32f1xx-hal.svg)](https://crates.io/crates/stm32f1xx-hal)
 [![Released API docs](https://docs.rs/stm32f1xx-hal/badge.svg)](https://docs.rs/stm32f1xx-hal)
 
-## Usage
+## Quick start guide
+
+Embedded rust development requires a bit more setup than ordinary development.
+For this guide, we'll assume you're using a stm32 blue pill board (shown
+below), but if you have another f1 microcontroller, you should be able to adapt
+it.
+
+![blue pill pinout](BluePillPinout.jpg "opt title")
+
+You will also need an [stlink](https://www.adafruit.com/product/2548) for
+programming and debugging. (There are many different versions of this out
+there, most of them _should_ work)
+
+### Installing software
+
+To program your microcontroller, you need to install:
+- [openocd](http://openocd.org/)
+- `arm-none-eabi-gdb`
+
+Finally, you need to install arm target support for the rust compiler. To do
+so, run
+```
+rustup target install thumbv7m-none-eabi
+```
+
+
+### Setting up your project
+
+Create a new rust project as you usually do with `cargo init`. The hello world
+of embedded development is usually to blink an LED and code to do so is
+available in [examples/blinky.rs](examples/blinky.rs). Copy that file to the
+`main.rs` of your project.
+
+You also need to add some dependencies to your `Cargo.toml`:
+
+```toml
+[dependencies]
+embedded-hal = "0.2.3"
+nb = "0.1.2"
+cortex-m = "0.6.2"
+cortex-m-rt = "0.6.11"
+# Panic behaviour, see https://crates.io/keywords/panic-impl for alternatives
+panic-halt = "0.2.0"
+
+[dependencies.stm32f1xx-hal]
+version = "0.5.2"
+features = ["rt", "stm32f103"]
+```
+
+If you build your project now, you should get a single error: `error: language item required, but not found: eh_personality`
+This unhelpful error message
+can be fixed by compiling for the right target. To do so, run
+```bash
+cargo build --target thumbv7m-none-eabi
+```
+
```

Comment:
Maybe also include a few words about putting that into a .cargo/config here? Can also do this as a follow up.